### PR TITLE
Adds ability to display images based on tag

### LINKED ISSUE #6: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [ ] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [ ] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [ ] Where in the code should we implement this functionality?
- [ ] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by jnogales319**
Closes #6

### CODE REVIEW

**Comment by jnogales319** [comment]
[Issue 6: Images by Tag](https://trello.com/c/wX0CmMoo/34-issue-6-images-by-tag)

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class ImagesController < ApplicationController
   def index
-    @images = Image.order(id: :desc)
+    @images = if params[:tag].present?
+                Image.tagged_with(params[:tag]).order(id: :desc)
```
Comment (blakeherrington) [inline app/controllers/images_controller.rb]: Good job finding `tagged_with`!

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+              else
+                Image.order(id: :desc)
```
Comment (blakeherrington) [inline app/controllers/images_controller.rb]: Here we have the `order(id: :desc)` duplicated. Can you think of a way we could reorganize this a little bit so that this is not repeated?

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -8,7 +8,6 @@ def setup
 
   def test_index__displays_create_new_link
     get images_url
-    assert_response :success
```
Comment (blakeherrington) [inline test/controllers/images_controller_test.rb]: Why was this line removed?

File: app/views/images/show.html.erb [hunk 4]
```diff
@@ -6,8 +6,10 @@
   <%= image_tag @image.url, class: 'js-image' %>
   <div>
     <strong>Tags:</strong>
```
Comment (blakeherrington) [inline app/views/images/show.html.erb]: This is minor (and from the previous story, so if you want to address it just do it in a separate commit), but instead of using <strong> we could use a bootstrap strap to bold the text.

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -36,6 +35,17 @@ def test_index__images_ordered_newest_to_oldest
     assert_equal([image3.id, image2.id, image1.id], image_tag_ids)
   end
 
+  def test_index__shows_images_with_queried_tag
+    Image.create!(url: @test_url)
+    Image.create!(url: @test_url, tag_list: @test_tag_list)
```
Comment (blakeherrington) [inline test/controllers/images_controller_test.rb]: Where is @test_tag_list defined? I don't see it in this commit. I'm guessing it's in the set up?

File: test/controllers/images_controller_test.rb [hunk 6]
```diff
@@ -36,6 +35,17 @@ def test_index__images_ordered_newest_to_oldest
...
+
+    get images_url(tag: 'mrTag')
+
+    response_images = css_select '.image__index_image'
+    assert_equal 1, response_images.count
+    assert_equal @test_url, response_images[0][:src]
```
Comment (blakeherrington) [inline test/controllers/images_controller_test.rb]: Does this test verify that the image with the correct tag is the one being shown?

File: app/controllers/images_controller.rb (hunk 1 above)
Comment (jnogales319) [reply app/controllers/images_controller.rb]: Why thank you

Comment (jnogales319) [reply app/controllers/images_controller.rb]: I could do something like the following:

```
@images = Image.order(id: :desc)
if params[:tag].present?
  @images = @images.tagged_with(params[:tag])
end
```

I'm not sure if this would open two database connections or if `.tagged_with` runs directly on the local data. Probably not an issue with the table size we're looking at, but would be a nice to know.

File: app/controllers/images_controller.rb (hunk 2 above)
Comment (jnogales319) [reply app/controllers/images_controller.rb]: I could do something like the following:

```
@images = Image.order(id: :desc)
if params[:tag].present?
  @images = @images.tagged_with(params[:tag])
end
```

I'm not sure if this would open two database connections or if `.tagged_with` runs directly on the local data. Probably not an issue with the table size we're looking at, but would be a nice to know.

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (jnogales319) [reply test/controllers/images_controller_test.rb]: The assert_response feels unnecessary to me as a successful response body is needed for `assert_select 'a[href=?]', new_image_url, count: 1` to pass. I can be convinced otherwise, of course.

File: test/controllers/images_controller_test.rb (hunk 5 above)
Comment (jnogales319) [reply test/controllers/images_controller_test.rb]: @test_tag_list was added in an earlier pull request and was/is used for `test_show__tag_is_shown` and `test_create__proper_url_succeeds`. See https://github.com/AppFolioOnboarding/image-sharer-jnogales319/commit/ce07ba4095295fe589b53842f5ae4aeca9796d6a

File: test/controllers/images_controller_test.rb (hunk 6 above)
Comment (jnogales319) [reply test/controllers/images_controller_test.rb]: It doesn't. It should.

File: app/views/images/show.html.erb (hunk 4 above)
Comment (jnogales319) [reply app/views/images/show.html.erb]: Taken care of in a new, separate commit
